server/budgettracker.business/Api/Contracts/BudgetApi/CreateBudgetRequestContract.cs

```diff
@@ -0,0 +1,35 @@
+using Newtonsoft.Json;
+
+namespace budgettracker.business.Api.Contracts.BudgetApi
+{
+    public class CreateBudgetRequestContract : IApiContract {
+
+        [JsonProperty("name")]
+        public string Name { get; set; }
+
+        [JsonProperty("set-amount")]
+        public decimal SetAmount { get; set; }
+
+        /// <summary>
+        /// <p> The duration of the budget cycle in days.</p>
+        /// </summary>    
+        [JsonProperty("duration")]
+        public int Duration { get; set; }
+
+        /// <summary>
+        /// <p> The guid the of the parent budget, if the budget is the 'root' 
+        /// this will be null. This string is converted to guid later. </p>
+        /// </summary>
+        [JsonProperty("parent-id")]
+        public string ParentId { get; set; }
+
+        public bool IsValid()
```

Comment:
That is fine I can pull it out into a static method. Just wanted to create something that checks if it is a valid request.

Reply:
Actually thinking about it maybe should be in this class since we would have to create a whole lot of IsValid methods. It could just live here so that it doesn't clutter up another file.

Reply:
Either way it's "cluttering" up some file. I guess we can leave it for now but as soon as we add other validation, let's move it to a new "BudgetValidation" file.